components/Typography/examples/TypographyMaterialScalableStylesViewController.h

```diff
@@ -0,0 +1,18 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+@interface TypographyMaterialScalableStyleViewController : UITableViewController
```

Comment:
Respectfully disagree. I'm trying to make this example as vanilla UIKit as possible, and that means a header w/ interface and source w/ implementation.

Reply:
I can understand the disagreement, but I believe in this case the project convention should be followed.   Unless otherwise necessary, we generally keep our examples in a single, self-contained `.m` file (or `.swift`). At the moment, the project has 187 `.m` files vs. 91 `.h` files in the `examples/` directories.  That indicates more than half of our examples files have no associated `.h` file.  Since some of those are reusable code, a number of `.h` files do make sense. However, in general we should stick to the overall convention.

Reply:
That convention has been painful.
This was mentioned at a recent meeting by a member of our team as well as outside clients.
If you feel strongly, I will change it, but with the understanding that it will be worse.

Reply:
Unfortunately I do not recall that meeting, but it would be good to document the problems it has caused.  I have not personally had any difficulties as a result of this convention, but it's helpful to make sure they're recorded and known by the team at large so we can make improvements.

Reply:
Additionally, this follows the pattern set by *every* other example in Typography.
Happy to discuss offline, but I'm not currently convinced that we should add a supplemental directory and break apart our example code into multiple files.

Reply:
Please do not create a `supplemental` directory, that's not my request. I'm suggesting that you put your interface in the `.m` file. This matches the majority of our examples throughout MDC-iOS.

Reply:
Headers exist for inclusion/import by the build system. In this case, we should not expect other code to import this controller. Since neither the Catalog nor Dragons requires importing headers to identify our examples, there is no need to have an extra file hanging around.

Reply:
Removed.